### PR TITLE
[11.0][mig] account_tax_analysis

### PR DESCRIPTION
**Pull request by sbidoul**
Look at the last 3 commits.

Migration plus a small improvement (which I will backport to 10 when this is merged).

### CODE REVIEW

File: account_tax_analysis/models/account_move_line.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+class AccountMoveLine(models.Model):
+
+    _inherit = 'account.move.line'
+
+    analysis_tax = fields.Char(
+        string="Analysis Tax", compute="_compute_analysis_tax", store=True)
+    account_type = fields.Many2one(
+        string="Account Type",
+        related='account_id.user_type_id', store=True)
```
Comment (rousseldenis) [inline account_tax_analysis/models/account_move_line.py]: @sbidoul Is this shouldn't be readonly?

Comment (sbidoul) [reply account_tax_analysis/models/account_move_line.py]: @rousseldenis done

**Review by rousseldenis — APPROVED** [review]
Code review

File: account_tax_analysis/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "version": "11.0.1.0.0",
+    "description": """
+        Add a report on tax (Invoicing / Reports / Taxes Analysis)""",
+    "depends": [
+        "base",
+        "account",
+    ],
+    "author": "Camptocamp SA, ACSONE SA/NV, Odoo Community Association (OCA)",
+    "category": 'Accounting & Finance',
+    "website": "http://www.camptocamp.com",
```
Comment (tbaden) [inline account_tax_analysis/__manifest__.py:16]: please change website to: https://github.com/OCA/account-financial-tools

File: account_tax_analysis/wizard/account_tax_analysis_view.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <menuitem
+        name="Taxes Analysis"
+        parent="account.menu_finance_reports"
+        id="menu_account_vat_declaration_analysis"/>
+
+    <menuitem
+        name="Taxes Analysis"
+        parent="menu_account_vat_declaration_analysis"
+        action="action_account_vat_declaration_analysis"
+        id="menu_action_account_vat_declaration_analysis"/>
```
Comment (tbaden) [inline account_tax_analysis/wizard/account_tax_analysis_view.xml:48]: id pattern (optional)
> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For a menu: <model_name>_menu

File: account_tax_analysis/wizard/account_tax_analysis_view.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        <field name="res_model">account.vat.declaration.analysis</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">form</field>
+        <field name="target">new</field>
+    </record>
+
+    <menuitem
+        name="Taxes Analysis"
+        parent="account.menu_finance_reports"
+        id="menu_account_vat_declaration_analysis"/>
```
Comment (tbaden) [inline account_tax_analysis/wizard/account_tax_analysis_view.xml:42]: id pattern (optional)
> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For a menu: <model_name>_menu

File: account_tax_analysis/wizard/account_tax_analysis_view.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                </group>
+                <footer>
+                    <button name="show_vat" string="Show" type="object" class="oe_highlight"/>
+                    <button special="cancel" string="Cancel" class="oe_link"/>
+                </footer>
+            </form>
+        </field>
+    </record>
+
+    <record id="action_account_vat_declaration_analysis" model="ir.actions.act_window">
```
Comment (tbaden) [inline account_tax_analysis/wizard/account_tax_analysis_view.xml:30]: id pattern (optional)
> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For an action: the main action respects <model_name>_action. Others are suffixed with _<detail>, where detail is  
> an underscore lowercase string explaining the action (should not be long). This is used only if multiple actions are declared for the model.

File: account_tax_analysis/wizard/account_tax_analysis_view.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Author: Vincent Renaville
+     Copyright 2013 Camptocamp SA
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <record id="view_account_vat_declaration_analysis" model="ir.ui.view">
```
Comment (tbaden) [inline account_tax_analysis/wizard/account_tax_analysis_view.xml:7]: id pattern (optional)

> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

File: account_tax_analysis/views/account_move_line.xml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                <button type="object" name="action_show_invoice" string="View invoice" icon="fa-search" attrs="{'invisible': [('invoice_id', '=', False)]}"/>
+                <field name="move_id"/>
+                <field name="date"/>
+                <field name="amount_currency" groups="base.group_multi_currency" />
+                <field name="currency_id" groups="base.group_multi_currency" />
+            </tree>
+        </field>
+    </record>
+
+    <record id="action_view_tax_analysis" model="ir.actions.act_window">
```
Comment (tbaden) [inline account_tax_analysis/views/account_move_line.xml:87]: id pattern (optional)

> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For an action: the main action respects <model_name>_action. Others are suffixed with _, where detail is
> an underscore lowercase string explaining the action (should not be long). This is used only if multiple actions are declared for the model.

File: account_tax_analysis/views/account_move_line.xml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                          context="{'group_by': 'date:month'}" />
+                    <filter name="group_by_year"
+                          string="Year"
+                          context="{'group_by': 'date:year'}" />
+                </group>
+            </search>
+        </field>
+    </record>
+
+    <record id="view_move_line_tree_tax_analysis" model="ir.ui.view">
```
Comment (tbaden) [inline account_tax_analysis/views/account_move_line.xml:63]: id pattern (optional)

> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

File: account_tax_analysis/views/account_move_line.xml [hunk 9]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Author: Vincent Renaville
+     Copyright 2013 Camptocamp SA
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <record id="view_account_move_line_filter_vat_analysis" model="ir.ui.view">
```
Comment (tbaden) [inline account_tax_analysis/views/account_move_line.xml:7]: id pattern (optional)

> Use the following patterns, where <model_name> is the name of the model that the menu, view, etc. belongs to (e.g. for a res.users form view, the name would be res_users_view_form):
> 
> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

**Review by tbaden — CHANGES_REQUESTED** [review]
thanks for the work 👍 
i did a code review. just a small change and some optional, nice to have, changes

**Pull request closed without merging**